src/site/content/en/blog/off-main-thread/index.md

```diff
@@ -0,0 +1,324 @@
+for moving a single piece of heavy work off the main thread.
+Trying to handle multiple operations with a single web worker gets unwieldy quickly:
+you have to encode not only the parameters but also the operation in the message,
+and you have to do bookkeeping to match responses to requests.
+That complexity is likely why web workers haven't been adopted more widely.
+
+But if we could remove some of the difficulty of communicating
+between the main thread and web workers,
+this model could be a great fit for many use cases.
+And, luckily, there's a library that does just that!
+
+## Comlink: making web workers less work
+
+[Comlink](http://npm.im/comlink) is a library
+whose goal is to let you use web workers
+without having to think about the details of `postMessage`.
+Comlink lets you to share variables
+between web workers and the main thread
+almost like programming languages that natively support threading.
+
+You set up Comlink by importing it in a web worker
+and defining a set of functions to expose to the main thread.
+You then import Comlink on the main thread, wrap the worker,
+and get access to the exposed functions:
+
+### `worker.js`
+```js
+import {expose} from "comlink";
+
+const api = {
+  someMethod() { /* … */ }
+}
+expose(api);
+```
+
+### `main.js`
+```js
+import {wrap} from "comlink";
+
+const worker = new Worker("./worker.js");
+const api = wrap(worker);
+```
+
+The `api` variable on main thread behaves the same as the one in the web worker,
+except that every function returns a promise for a value rather than the value itself.
+
+## What code should you move to a web worker?
+
+Web workers don't have access to the DOM and many APIs
+like [WebUSB](https://developer.mozilla.org/en-US/docs/Web/API/USB),
+[WebRTC](https://developer.mozilla.org/en-US/docs/Web/API/WebRTC_API), or
+[Web Audio](https://developer.mozilla.org/en-US/docs/Web/API/Web_Audio_API),
+so you can't put pieces of your app that rely on such access in a worker.
+Still, every small piece of code moved to a worker buys more headroom
+on the main thread for stuff that _has_ to be there—like updating the user interface.
+
+{% Aside %}
+Restricting UI access  to the main thread is actually typical in other languages.
+In fact, both iOS and Android call the main thread the _UI thread_.
+{% endAside %}
+
+One problem for web developers is that most web apps rely on a UI framework
+like Vue or React to orchestrate everything in the app;
+everything is a component of the framework and so is inherently tied to the DOM.
+That would seem to make it difficult to migrate to an OMT architecture.
+
+However, if we shift to a model in which UI concerns are separated from other concerns,
+like state management, web workers can be quite useful even with framework-based apps.
+That's exactly the approach taken with PROXX.
+
+## PROXX: an OMT case study
+
+The Google Chrome team developed [PROXX](https://github.com/GoogleChromeLabs/proxx)
```

Comment:
```suggestion
The Google Chrome team developed [PROXX](/load-faster-like-proxx/)
```

I'm going to give our other PROXX article a plug here. Perhaps not the best link, but it's good enough because the first paragraph conveys the basic idea of the app.